lib/plugin.js

```diff
@@ -115,7 +115,13 @@ internals.schema.strategy = Joi.object({
         timeSkewSec: Joi.number().integer().min(0).default(0)
     })
         .when('.validate', { is: Joi.not(false), then: Joi.allow(false) })
-        .required()
+        .required(),
+
+    headerName: Joi.alternatives()
+        .try(Joi.string(), Joi.boolean()).default('authorization').optional(),
```

Comment:
Why allow booleans?

Reply:
I allowed booleans so that we can specify false to not allow reading from headerName or cookieName. So by default it reads from header named `authorization` so if we specify another string then we will read from that. Then if we need to be able to read from cookie we need to make it `false` to skip reading from headerName and that will read from cookieName. This is how we can specify whether to read from only header or only cookie. Otherwise check the header first then check the cookie.

Reply:
This won't work since `true` has no meaning. It is better specified without alternatives as:

```js
Joi.string().allow(false).default('authorization')
```

Note I also removed the redundant `optional()` specifier, which is implied.